matter_server/server/device_controller.py

```diff
@@ -114,22 +112,15 @@ async def commission_with_code(self, code: str) -> MatterNode:
         """
         node_id = self._get_next_node_id()
 
-        # TODO TEMP !!!
-        # The call to CommissionWithCode returns early without waiting ?!
-        # This is most likely a bug in the SDK or its python wrapper
-        # success = await self._call_sdk(
-        #     self.chip_controller.CommissionWithCode,
-        #     setupPayload=code,
-        #     nodeid=node_id,
-        # )
-        # if not success:
-        #     raise NodeCommissionFailed(f"Commission with code failed for node {node_id}")
-        await self._call_sdk(
+        success = await self._call_sdk(
             self.chip_controller.CommissionWithCode,
             setupPayload=code,
             nodeid=node_id,
         )
-        await asyncio.sleep(60)
+        if not success:
+            raise NodeCommissionFailed(
+                f"Commission with code failed for node {node_id}"
```

Comment:
Side note: There's no way to get more context why the commissioning failed?

Reply:
No, unfortunately not. The error is not exposed. Good news is that this is changed upstream in the sdk code so in matter 1.1 release we will have more context